src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -471,6 +444,82 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
+func redpandaEnvVars() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "REDPANDA_ENVIRONMENT",
+			Value: "kubernetes",
+		},
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.name",
+				},
+			},
+		},
+		{
+			Name: "POD_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name: "POD_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "status.podIP",
+				},
+			},
+		},
+	}
+}
+
+func (r *StatefulSetResource) wasmContainer() corev1.Container {
+	if r.pandaCluster.Spec.DeepCopy().ResourcesWASM == nil {
+		r.logger.Info("BUG! No resources found for wasm - this should never happen with defaulting webhook enabled")
+		r.pandaCluster.Spec.ResourcesWASM = redpandav1alpha1.DefaultWASMResources
+	}
+	return corev1.Container{
+		Name:      wasmContainerName,
+		Image:     r.pandaCluster.FullImageName(),
+		Command:   []string{"/usr/local/bin/coproc.sh"},
+		Env:       redpandaEnvVars(),
+		Resources: *r.pandaCluster.Spec.ResourcesWASM,
+		VolumeMounts: append([]corev1.VolumeMount{
+			{
+				Name:      "config-dir",
+				MountPath: configDestinationDir,
+			},
+		}, r.secretVolumeMounts()...),
+	}
+}
+
+func (r *StatefulSetResource) rpkStatusContainer() corev1.Container {
+	if r.pandaCluster.Spec.ResourcesRpkStatus == nil {
+		r.logger.Info("BUG! No resources found for rpk status - this should never happen with defaulting webhook enabled")
+		r.pandaCluster.Spec.ResourcesRpkStatus = redpandav1alpha1.DefaultRpkResources
+	}
+	return corev1.Container{
+		Name:      rpkStatusContainerName,
+		Image:     r.pandaCluster.FullImageName(),
+		Command:   []string{"/usr/local/bin/rpk-status.sh"},
+		Env:       redpandaEnvVars(),
```

Comment:
I think only `REDPANDA_ENVIRONMENT` is necessary here, as we report this back to the `metrics` service.

REF: https://github.com/vectorizedio/redpanda/pull/567